src/algebra/homology/connective_chain_complex.lean

```diff
@@ -0,0 +1,494 @@
+    dsimp,
+    split_ifs,
+    { simp only [category.assoc, eq_to_hom_trans],
+      rw [←homological_complex.eq_to_hom_f _ (int.to_nat_of_nonneg h)],
+      simp, },
+    { simp at h, simp [h], }
+  end, }.
+
+/-- The inverse of the counit for the equivalence. -/
+@[simps]
+def counit_inv : 𝟭 _ ⟶ to_connective_chain_complex V ⋙ to_chain_complex V :=
+{ app := λ C,
+  { f := λ i,
+    if h : 0 ≤ i then eq_to_hom (by simp [X_ℤ, if_pos h, int.to_nat_of_nonneg h]) else
+      (iso_of_is_isomorphic_zero (C.property i (show i < 0, by simpa using h))).hom ≫
+        eq_to_hom (show 0 = ite (0 ≤ i) (C.val.X (i.to_nat)) 0, by simp [if_neg h]),
+    comm' :=
+    begin
+      ext i, dsimp [d_ℤ],
+      by_cases h : 0 ≤ i,
+      { by_cases h' : 0 ≤ i - 1,
+        { dsimp [X_ℤ],
+          have h'' : 0 < i := by linarith,
+          simp only [dif_pos h, dif_pos h', dif_pos h''],
+          simp only [category.id_comp, category.assoc, eq_to_hom_trans, eq_to_hom_trans_assoc],
+          erw homological_complex.eq_to_hom_d_assoc C.val
+            (show i = ↑(i.to_nat - 1) + 1, by simp [h'']),
+          simp, refl, },
+        { rw [dif_pos h, dif_neg h'],
+          have h'' : i = 0 := by linarith,
+          subst h'',
+          simp, }, },
+      { have h' : ¬ 0 ≤ i - 1 := by linarith,
+        have h'' : ¬ 0 < i := by linarith,
+        rw [dif_neg h, dif_neg h', dif_neg h''],
+        simp at h'',
+        simp [h''], },
+    end, },
+  naturality' := λ C D f,
+  begin
+    ext i,
+    dsimp,
+    split_ifs,
+    { simp only [eq_to_hom_trans_assoc],
+      rw [homological_complex.eq_to_hom_f_assoc _ (int.to_nat_of_nonneg h).symm],
+      simp, },
+    { simp at h, simp [h], }
+  end, }.
+
+/--
+The counit isomorphism for the equivalence
+`connective_chain_complex V ≌ { C : chain_complex V // is_connective C }`.
+-/
+@[simps]
+def counit_iso :
+  to_connective_chain_complex V ⋙ to_chain_complex V ≅ 𝟭 _ :=
+{ hom := counit_hom V,
+  inv := counit_inv V,
+  hom_inv_id' := by { ext C i, dsimp, split_ifs; simp, },
+  inv_hom_id' := begin
+    ext C i,
+    dsimp,
+    split_ifs with h,
+    { simp, },
+    { simp at h, simp [h], }
+  end, }.
+
+lemma functor_unit_iso_comp (C : connective_chain_complex V) :
+  (to_chain_complex V).map ((unit_iso V).hom.app C) ≫
+      (counit_iso V).hom.app ((to_chain_complex V).obj C) =
+    𝟙 ((to_chain_complex V).obj C) :=
+begin
+  ext i,
+  dsimp,
+  split_ifs,
+  { simp only [category.id_comp, eq_to_hom_refl, eq_to_hom_trans],
+    simp [if_pos h], refl, },
+  { dsimp [X_ℤ],
+    simp only [limits.zero_comp],
+    refine (zero_of_target_iso_zero _ _).symm,
+    simp [if_neg h], },
+end
+
+end equivalence
+open equivalence
+
+/--
+The equivalence between `ℕ`-indexed chain complexes (with `d n : C (n+1) ⟶ C n`)
+and `ℤ`-indexed chain complexes supported in non-negative degrees (with `d i : C i ⟶ C (i-1)`).
+-/
+@[simps]
+def equivalence : connective_chain_complex V ≌ { C : chain_complex V // is_connective C } :=
```

Comment:
Is this a case where we don't care about what the unit/counit isomorphisms are and you can get away with using `equivalence.mk`? Or do we genuinely need to prove `functor_unit_iso_comp`?